server/test/index.test.js

```diff
@@ -17,29 +17,155 @@
+const TIMEOUT = 3000;
 
 describe('index.js tests', () => {
     after(() => {
         mock.stop('open');
     });
-    it('Just calling index', async function () {
+    afterEach(() => {
+        delete process.env.ASSET_COMPUTE_DEV_TOOL_ENV;
+    });
+    it("devtool starts and serves html", async function() {
+        // set up server
         stdout.start();
-        await require('../index.js');
+        const devtool = new DevtoolServer();
+        await devtool.run();
+        await sleep(SERVER_START_UP_WAIT_TIME);
+        const port = devtool.port;
         stdout.stop();
+        
+        // check start up logs
+        const stdoutList = stdout.output.split('\n');
+        assert(stdoutList[0].includes(`Asset Compute Developer Tool Server started on url http://localhost:${port}/?devToolToken=`));
+        const url = stdoutList[0].split(' ').pop();
+        assert.ok(url.includes(`http://localhost:${port}/?devToolToken=`));
+        
+        // api call to get raw html
+        const resp = await fetch(url);
+        assert.strictEqual(resp.status, 200);
+        const html = await resp.text();
+        assert.ok(html.includes('/static/js'));
+        await devtool.stop();
+    });
 
+    it("devtool starts and serves html in development mode", async function() {
+        process.env.ASSET_COMPUTE_DEV_TOOL_ENV = 'development';
+        // set up server
+        stdout.start();
+        const devtool = new DevtoolServer();
+        await devtool.run();
+        await sleep(SERVER_START_UP_WAIT_TIME);
+        const port = devtool.port;
+        stdout.stop();
+        
         // check start up logs
-        console.log(stdout.output);
-        assert.ok(stdout.output.includes('client-build'));
+        assert.ok(stdout.output.includes('Asset Compute Developer Tool Server started. Running in development mode.'));
+        
+        // api call to get raw html
+        const resp = await fetch(`http://localhost:${port}`);
+        assert.strictEqual(resp.status, 200);
+        const html = await resp.text();
+        assert.ok(html.includes('/static/js'));
+        await devtool.stop();
+    });
+
+
+    it("server starts up and does an api call", async function() {
+        this.timeout(TIMEOUT);
+        // set up server
+        stdout.start();
+        const devtool = new DevtoolServer();
+        await devtool.run();
+        await sleep(SERVER_START_UP_WAIT_TIME);
+
+        // check output
+        stdout.stop();
+        const port = devtool.port;
+        const stdoutList = stdout.output.split('\n');
+        const url = stdoutList[0].split(' ').pop();
+        const token = url.split('=')[1];
+        assert.strictEqual(token.length, 64);
+        assert.ok(url.includes(`http://localhost:${port}/?devToolToken=`));
+        
+        // api call to get raw html
+        const resp = await fetch(`http://localhost:${port}/api/asset-compute-endpoint`, {
+            headers: {
+                "authorization": token,
```

Comment:
Nitpick: [Authorization is upper-case](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization).
```suggestion
                "Authorization": token
```